gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +967,920 @@ public void testListDnsRecords() {
+      assertNotNull(created.startTimeMillis());
+      assertTrue(created.deletions().isEmpty());
+      assertEquals("7", created.generatedId());
+      assertNull(created.status());
+      waitForChangeToComplete(created);
+      created = DNS.applyChangeRequest(ZONE1.name(), CHANGE_DELETE_ZONE1);
+      waitForChangeToComplete(created);
+      batch = DNS.batch();
+      result = batch.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1,
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.ADDITIONS));
+      batch.submit();
+      created = result.get();
+      assertEquals(CHANGE_ADD_ZONE1.additions(), created.additions());
+      assertNull(created.startTimeMillis());
+      assertTrue(created.deletions().isEmpty());
+      assertEquals("9", created.generatedId());
+      assertNull(created.status());
+      // finishes with delete otherwise we cannot delete the zone
+      waitForChangeToComplete(created);
+      batch = DNS.batch();
+      result = batch.applyChangeRequest(ZONE1.name(), CHANGE_DELETE_ZONE1,
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.DELETIONS));
+      batch.submit();
+      created = result.get();
+      waitForChangeToComplete(created);
+      assertEquals(CHANGE_DELETE_ZONE1.deletions(), created.deletions());
+      assertNull(created.startTimeMillis());
+      assertTrue(created.additions().isEmpty());
+      assertEquals("10", created.generatedId());
+      assertNull(created.status());
+      waitForChangeToComplete(created);
+    } finally {
+      clear();
+    }
+  }
+
+  @Test
+  public void testGetChangeBatch() {
+    try {
+      Zone zone = DNS.create(ZONE1, Dns.ZoneOption.fields(Dns.ZoneField.NAME));
+      ChangeRequest created = zone.applyChangeRequest(CHANGE_ADD_ZONE1);
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<ChangeRequest> result =
+          batch.getChangeRequest(zone.name(), created.generatedId());
+      batch.submit();
+      assertEqChangesIgnoreStatus(created, result.get());
+      waitForChangeToComplete(zone.name(), created.generatedId());
+      zone.applyChangeRequest(CHANGE_DELETE_ZONE1);
+      // with options
+      created = zone.applyChangeRequest(CHANGE_ADD_ZONE1,
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.ID));
+      batch = DNS.batch();
+      result = batch.getChangeRequest(zone.name(), created.generatedId(),
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.ID));
+      batch.submit();
+      assertEqChangesIgnoreStatus(created, result.get());
+      waitForChangeToComplete(zone.name(), created.generatedId());
+      zone.applyChangeRequest(CHANGE_DELETE_ZONE1);
+      created = zone.applyChangeRequest(CHANGE_ADD_ZONE1,
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.STATUS));
+      batch = DNS.batch();
+      result = batch.getChangeRequest(zone.name(), created.generatedId(),
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.STATUS));
+      batch.submit();
```

Comment:
If waiting does take a long time, you could alternate additions and deletions like you did above to cut the number of waits in half.  Otherwise, probably not worth it.